src/rgw/rgw_op.cc

```diff
@@ -3304,6 +3325,33 @@ void RGWCreateBucket::execute()
 
   op_ret = store->ctl()->bucket->link_bucket(s->user->get_id(), s->bucket->get_bi(),
                                           s->bucket->get_creation_time(), s->yield, false);
+  if (!bucket_count_initialized) {
+    rgw::sal::RGWBucketList buckets;
+    string marker;
+    int ret = rgw_read_user_buckets(store, s->user->get_id(), buckets,
+				    marker, string(), s->user->get_max_buckets(),
+				    false);
```

Comment:
Ack. We are initializing bucket_count irrespective of s->user->get_max_buckets() value here. This will be one time when $user.buckets is available for write and also when first bucket gets created or we have upgraded cluster where bucket_count is yet to be initialized. It does makes sense to use bucket count if calculated in `verify_permission()`.